image/DockerfileBase

```diff
@@ -0,0 +1,151 @@
+    git log -1 --format='%H'; \ 
+    mkdir -p build; \
+    cd build; \
+    cmake ..  -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/apps/pocl/4.0-llvm-${LLVM_VERSION} -DENABLE_TESTS=OFF -DENABLE_EXAMPLES=OFF -DSTATIC_LLVM=ON -DKERNELLIB_HOST_CPU_VARIANTS=distro; \
+    cmake --build . --parallel 4; \
+    sudo cmake --install . ; \
+    sudo mkdir -p /apps/modulefiles/Core/pocl/; \
+    echo "prepend_path(\"CPATH\",\"/apps/llvm/${LLVM_VERSION}/include\")" | sudo tee  /apps/modulefiles/Core/pocl/4.0-llvm-${LLVM_VERSION}.lua; \
+    echo "prepend_path(\"LD_LIBRARY_PATH\",\"/apps/pocl/4.0-llvm-${LLVM_VERSION}/lib\")" | sudo tee -a /apps/modulefiles/Core/pocl/4.0-llvm-${LLVM_VERSION}.lua; \
+    echo "prepend_path(\"LIBRARY_PATH\",\"/apps/llvm/${LLVM_VERSION}/lib\")" | sudo tee -a /apps/modulefiles/Core/pocl/4.0-llvm-${LLVM_VERSION}.lua; \
+    echo "prepend_path(\"PATH\",\"/apps/pocl/4.0-llvm-${LLVM_VERSION}/bin\")" | sudo tee -a /apps/modulefiles/Core/pocl/4.0-llvm-${LLVM_VERSION}.lua; \
+    echo "prepend_path(\"XDG_DATA_DIRS\",\"/apps/pocl/4.0-llvm-${LLVM_VERSION}/share\")" | sudo tee -a /apps/modulefiles/Core/pocl/4.0-llvm-${LLVM_VERSION}.lua; \
+    echo "setenv(\"OPENCL_VENDOR_PATH\",\"/apps/pocl/4.0-llvm-${LLVM_VERSION}/etc/OpenCL/vendors\")" | sudo tee -a /apps/modulefiles/Core/pocl/4.0-llvm-${LLVM_VERSION}.lua; \
+    echo "load(\"llvm/${LLVM_VERSION}\", \"pocl/4.0-llvm-${LLVM_VERSION}\")" | sudo tee  /apps/modulefiles/Core/StdEnv.lua
+#////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+#//                                                                                                                                        //
+#//                                                         End of base image                                                              //
+#//                                                                                                                                        //
+#////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+
+
+# RUN sudo apt install -y gpg-agent wget; \
+#     wget -O- https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB | gpg --dearmor | sudo tee /usr/share/keyrings/oneapi-archive-keyring.gpg > /dev/null; \
+#     echo "deb [signed-by=/usr/share/keyrings/oneapi-archive-keyring.gpg] https://apt.repos.intel.com/oneapi all main" | sudo tee /etc/apt/sources.list.d/oneAPI.list; \
+#     sudo apt update; \
+#     sudo apt install -y intel-oneapi-compiler-dpcpp-cpp intel-oneapi-runtime-opencl
+
+# RUN git clone https://github.com/intel/compute-runtime neo; \
+#     cd neo; mkdir build; cd build;\
+#     cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/apps/intel -DNEO_SKIP_UNIT_TESTS=1 ../; \
+#     make -j 4; \
+#     sudo make install
```

Comment:
Here is what I used to create a virtual conda environment and installing oneapi:

```
    1  ml
    2  ls
    3  pwd
    4  rm -rf ./*
    5  ls
    6  # Download and install Miniconda
    7  MINICONDA_INSTALLER=Miniconda3-latest-Linux-x86_64.sh
    8  wget https://repo.anaconda.com/miniconda/$MINICONDA_INSTALLER -O $MINICONDA_INSTALLER
    9  bash $MINICONDA_INSTALLER -b -p $HOME/miniconda
   10  rm $MINICONDA_INSTALLER
   11  # Initialize conda
   12  source $HOME/miniconda/bin/activate
   13  conda init
   14  # Add Intel channel
   15  conda config --add channels intel
   16  # Add Intel channel
   17  conda config --add channels intel
   18  # Create environment called oneapi-2024.1 and install packages
   19  conda create -n oneapi-2024.1 -y intel-oneapi-compilers intel-oneapi-mkl intel-oneapi-runtime intel-oneapi-opencl intel-oneapi-dpcpp-cpp intel-oneapi-dpcpp-cpp-compiler
   20  conda install -y     dpcpp-cpp-rt=2024.1.0=intel_963     dpcpp_impl_linux-64=2024.1.0=intel_963     icc_rt=2024.1.0=intel_963     intel-cmplr-lib-rt=2024.1.0=intel_963     intel-cmplr-lic-rt=2024.1.0=intel_963     intel-opencl-rt=2024.1.0=intel_963     intel-openmp=2024.1.0=intel_963     mkl=2024.1.0=intel_691     mkl-devel=2024.1.0=intel_691     mkl-devel-dpcpp=2024.1.0=intel_691     mkl-dpcpp=2024.1.0=intel_691     mkl-include=2024.1.0=intel_691     onedpl-devel=2022.5.0=intel_215     onemkl-sycl-blas=2024.1.0=intel_691     onemkl-sycl-datafitting=2024.1.0=intel_691     onemkl-sycl-dft=2024.1.0=intel_691     onemkl-sycl-lapack=2024.1.0=intel_691     onemkl-sycl-rng=2024.1.0=intel_691     onemkl-sycl-sparse=2024.1.0=intel_691     onemkl-sycl-stats=2024.1.0=intel_691     onemkl-sycl-vm=2024.1.0=intel_691     tbb=2021.12.0=intel_495
   21  # Create environment called oneapi-2024.1
   22  conda create -n oneapi-2024.1 -y
   23  # Activate the new environment
   24  conda activate oneapi-2024.1
   25  conda install -y     dpcpp-cpp-rt=2024.1.0=intel_963     dpcpp_impl_linux-64=2024.1.0=intel_963     icc_rt=2024.1.0=intel_963     intel-cmplr-lib-rt=2024.1.0=intel_963     intel-cmplr-lic-rt=2024.1.0=intel_963     intel-opencl-rt=2024.1.0=intel_963     intel-openmp=2024.1.0=intel_963     mkl=2024.1.0=intel_691     mkl-devel=2024.1.0=intel_691     mkl-devel-dpcpp=2024.1.0=intel_691     mkl-dpcpp=2024.1.0=intel_691     mkl-include=2024.1.0=intel_691     onedpl-devel=2022.5.0=intel_215     onemkl-sycl-blas=2024.1.0=intel_691     onemkl-sycl-datafitting=2024.1.0=intel_691     onemkl-sycl-dft=2024.1.0=intel_691     onemkl-sycl-lapack=2024.1.0=intel_691     onemkl-sycl-rng=2024.1.0=intel_691     onemkl-sycl-sparse=2024.1.0=intel_691     onemkl-sycl-stats=2024.1.0=intel_691     onemkl-sycl-vm=2024.1.0=intel_691     tbb=2021.12.0=intel_495
   26  clinfo
   27  clear
   28  clinfo -l
   29  which icpx
   30  history
(oneapi-2024.1) chipStarUser@b57ec91c4164:~$
```

from there, you can create a module to load it:

```
╭─pvelesko@cupcake ~
╰─$ module show oneapi/2024.1.0                                                                                                                                                                                                                                                                                                                                       130 ↵
-------------------------------------------------------------------
/home/pvelesko/modulefiles/oneapi/2024.1.0:

prereq          opencl/ocl-icd-loader
prepend-path    PATH /home/pvelesko/miniconda3/envs/oneapi-2024.1.0/bin
prepend-path    LD_LIBRARY_PATH /home/pvelesko/miniconda3/envs/oneapi-2024.1.0/lib
prepend-path    LD_LIBRARY_PATH /home/pvelesko/miniconda3/envs/oneapi-2024.1.0/lib64
prepend-path    LIBRARY_PATH /home/pvelesko/miniconda3/envs/oneapi-2024.1.0/lib
prepend-path    LIBRARY_PATH /home/pvelesko/miniconda3/envs/oneapi-2024.1.0/lib64
prepend-path    PKG_CONFIG_PATH /home/pvelesko/miniconda3/envs/oneapi-2024.1.0/lib/pkgconfig
prepend-path    CMAKE_PREFIX_PATH /home/pvelesko/miniconda3/envs/oneapi-2024.1.0/lib/cmake
-------------------------------------------------------------------
```